src/pgroonga.c

```diff
@@ -6119,6 +6127,14 @@ PGrnSearchBuildCondition(Relation index, ScanKey key, PGrnSearchData *data)
 		PGrnSearchBuildConditionIn(data, key, targetColumn, attribute);
 		return;
 	}
+	if (PGrnSearchIsMatchInCondition(key))
+	{
+		// PostgreSQL 18 optimaize to "column IN (keyword1, keyword2, ...)" from
+		// "column %% keyword1 OR column %% keyword2 OR ...".
+		// %% is match operator. So, we handle "column %% keyword1 OR column %%
+		// keyword2 OR ..." as "match in" operator.
+		key->sk_strategy = PGrnMatchInStrategyV2Number;
```

Comment:
How about adding a PGrnSearchBuildMatchConditionIn() like PGrnSearchIsInCondition() and PGrnSearchBuildConditionIn()?

Reply:
I added a `PGrnSearchBuildConditionMatchIn()`.

Reply:
What do you think of renaming this function to `PGrnSearchSetMatchInStrategy()` ?
Regarding `PGrnSearchBuildConditionMatchIn()`, the function name suggests that it builds a condition.
But it only modifies the parameters (specifically, the sk_strategy). Compared to other functions in the `BuildCondition` family, its behavior is inconsistent.

Reply:
Thank you for your comment.
I've fixed it.